src/utils.ts

```diff
@@ -1,32 +1,28 @@
-import {
-  isEqual,
-  omit,
-  keys,
-  map,
-  find,
-  reduce
-} from 'lodash';
+import { find, isEqual, map, reduce } from 'lodash';
+import { HistogramValue, Labels, Metric, MetricValue } from './collector';
 
-
-function getLabelPairs(metric) {
-  const pairs = map(omit(metric, 'value'), (v, k) => `${k}="${v}"`);
+function getLabelPairs(metric: Metric<MetricValue>) {
+  const pairs = map(metric.labels, (v, k) => `${k}="${v}"`);
   return pairs.length === 0 ? '' : `${pairs.join(',')}`;
 }
 
-export function formatHistogramOrSummary(name, metric, bucketLabel = 'le') {
+export function formatHistogramOrSummary(
+  name: string,
+  metric: Metric<HistogramValue>,
+  bucketLabel = 'le',
+): string {
   let str = '';
   const labels = getLabelPairs(metric);
-  if (labels) {
+  if (labels.length > 0) {
```

Comment:
suggest trying lodash.isEmpty

Reply:
it adds an overhead and as long as we have the types here user should carefully passing variables from JS-code and perform type checking in his code. Do you agree?
If we assume that user can pass anything to library code, we should wrap all those variables in checks like `lodash.isEmpty` and it seems to be the madness.

Reply:
I'm very sympathetic to that line of thinking.  I understand where you're coming from but I happen to take the (extreme, I know) position that we should do everything we reasonably can to prevent simple failures of this sort - including checking all variables a user can pass in.  I've seen it done both ways and although it adds a lot of boilerplate to be comprehensive about checking this kind of thing everywhere, it ends up being (at least) very obvious what's going on, which is nice.

Feel free to ignore. :)  Thanks so much for the contributions!

Reply:
I found the better solution — I check runtime types in `Registry.create` function, e.g. on input.
In this case our internal code will be always type safe.